src/clipboard/mod.rs

```diff
@@ -1,100 +1,74 @@
 //! Provides a clipboard abstraction to access the target system's clipboard.
 
 use copypasta::{ClipboardContext, ClipboardProvider};
-use std::fmt;
+use dioxus::prelude::{RefCell, ScopeState};
+use std::rc::Rc;
 
-/// Contains the context for interacting with the clipboard.
+pub fn use_init_clipboard(cx: &ScopeState) {
+    cx.use_hook(|| {
+        if let Ok(clipboard) = ClipboardContext::new() {
+            cx.provide_context(Rc::new(RefCell::new(clipboard)));
```

Comment:
We can remove the use_init method here by trying to read the context and then inserting the context at the root if we don't find it. Like the signals crate does [here](https://github.com/DioxusLabs/dioxus/blob/master/packages/signals/src/rt.rs#L13). Then you can just use use_clipboard directly without worrying about the context

Reply:
> We can remove the use_init method here by trying to read the context and then inserting the context at the root if we don't find it. Like the signals crate does [here](https://github.com/DioxusLabs/dioxus/blob/master/packages/signals/src/rt.rs#L13). Then you can just use use_clipboard directly without worrying about the context

Can we though? Last I tried that, it didn't work in nested components. I think it got fixed on the master branch but not released. I might be wrong, I'll try it later

Reply:
Nevermind, it seems to work just fine... weird 🤔 I'll revisit some of my hooks in freya then